src/renderer/display_list.rs

```diff
@@ -147,6 +142,16 @@ impl<'a> DisplayList<'a> {
         }
         Ok(())
     }
+
+    fn get_max_lineno(&self) -> Option<usize> {
+        let max_lineno = self.body.iter().fold(None, |max, set| {
```

Comment:
As this only needs state for getting `self.body`, let's make this a free function (ie not in an `impl` block and not taking `&self`)

Reply:
I don't understand why a free function would be better than a method in this case, but I've made it